lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -1,22 +1,35 @@
 require 'hammer_cli_foreman/api/session_authenticator_wrapper'
+require 'hammer_cli_foreman/api/authenticator'
 require 'hammer_cli_foreman/api/interactive_basic_auth'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth_via_2fa'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth'
 require 'hammer_cli_foreman/api/void_auth'
+require 'uri'
 
 module HammerCLIForeman
+  AUTH_TYPES = {
+    basic_auth: "Basic_Auth",
+    oauth_with_2fa: "Oauth_With_Two_Factor",
+    oauth_without_2fa: "Oauth_Without_Two_Factor"
+  }.freeze
+
+  CONNECTION_NAME = 'foreman'
+
   module Api
     class Connection < HammerCLI::Apipie::ApiConnection
       attr_reader :authenticator
 
-      def initialize(settings, logger = nil, locale = nil)
-        default_params = build_default_params(settings, logger, locale)
+      def initialize(settings, logger = nil, locale = nil, auth_type = nil)
+        auth_type ||= default_auth_type(settings)
+        default_params = build_default_params(settings, logger, locale, auth_type)
         super(default_params,
           :logger => logger,
           :reload_cache => settings.get(:_params, :reload_cache) || settings.get(:reload_cache)
         )
       end
 
       def login
-        # Call some api entry point to trigger the
+        # Call some api entry point to trigger the successful connection
         @api.resource(:home).action(:status).call
```

Comment:
It would be worth considering to choose different call. The status became quite expensive recently.  It touches a lot of resources especially when Katello is around. :auth_statuses :index looked promising or :architectures.

Reply:
@ofedoren you are right I didn't noticed there are two endpoints and was mislead by other error.